### PR TITLE
HaskellLintBear.py: Import CabalRequirement

### LINKED ISSUE #1358: HaskellLintBear should have a CabalRequirement
CabalRequirement (and DistributionRequirement) was created recently. It should be used for HaskellLintBear.

### PR DESCRIPTION
**Pull request by ankit01ojha**
Closed #1358

### CODE REVIEW

File: bears/haskell/HaskellLintBear.py [hunk 1]
```diff
@@ -19,7 +23,12 @@ class HaskellLintBear:
     """
 
     LANGUAGES = {'Haskell'}
-    REQUIREMENTS = {DistributionRequirement(apt_get='hlint')}
+    REQUIREMENTS = {AnyOneOfRequirements(
+            [CabalRequirement(package='hlint', version='1.9.35'),
+             DistributionRequirement(apt_get='hlint')
+             ]
+        ),
```
Comment (jayvdb) [inline bears/haskell/HaskellLintBear.py]: please fix the alignment of open & closes of brackets.

**Comment by ankit01ojha** [comment]
@jayvdb , modified!

**Comment by jayvdb** [comment]
Please update your commit message to follow https://coala.io/commit

**Comment by ankit01ojha** [comment]
@jayvdb , I don't know why the tests are failing. 1st time all the tests got passed.

**Comment by manankalra** [comment]
@ankit01ojha Please change your shortlog to something like **Use CabalRequirement**. Accordingly, change the commit body too.

File: bears/haskell/HaskellLintBear.py [hunk 2]
```diff
@@ -45,7 +54,7 @@ def process_output(self, output, filename, file):
             for other_lines in range(1, len(from_lines)):
                 assert from_lines[other_lines] == to_lines[other_lines]
             line_nr = issue['startLine']
-            line_to_change = file[line_nr-1]
+            line_to_change = file[line_nr - 1]
```
Comment (manankalra) [inline bears/haskell/HaskellLintBear.py]: unnecessary change

File: bears/haskell/HaskellLintBear.py [hunk 3]
```diff
@@ -19,7 +23,12 @@ class HaskellLintBear:
     """
 
     LANGUAGES = {'Haskell'}
-    REQUIREMENTS = {DistributionRequirement(apt_get='hlint')}
+    REQUIREMENTS = {AnyOneOfRequirements(
+        [CabalRequirement(package='hlint', version='1.9.35'),
```
Comment (manankalra) [inline bears/haskell/HaskellLintBear.py]: identation
this is better:
```
REQUIREMENTS = {AnyOneOfRequirements(
        [CabalRequirement(package='hlint', version='1.9.35'),
         DistributionRequirement(apt_get='hlint')
         ]
    ),
}
```

File: bears/haskell/HaskellLintBear.py (hunk 2 above)
Comment (ankit01ojha) [reply bears/haskell/HaskellLintBear.py]: @manankalra Since, coala PEP8 , this is a suitable change.

File: bears/haskell/HaskellLintBear.py (hunk 3 above)
Comment (ankit01ojha) [reply bears/haskell/HaskellLintBear.py]: @manankalra , i don't see any issue with this, as it still follows PEP8.

File: bears/haskell/HaskellLintBear.py (hunk 2 above)
Comment (manankalra) [reply bears/haskell/HaskellLintBear.py]: but it isn't related to the issue you're working on
_different issue, different commit_ - not here ;)

File: bears/haskell/HaskellLintBear.py [hunk 4]
```diff
@@ -19,7 +23,12 @@ class HaskellLintBear:
     """
 
     LANGUAGES = {'Haskell'}
-    REQUIREMENTS = {DistributionRequirement(apt_get='hlint')}
+    REQUIREMENTS = {AnyOneOfRequirements(
```
Comment (jayvdb) [inline bears/haskell/HaskellLintBear.py]: put `Any...` on the next line indented, and keep indenting for each new level, and always add a trailing comma.

File: bears/haskell/HaskellLintBear.py (hunk 3 above)
Comment (jayvdb) [reply bears/haskell/HaskellLintBear.py]: @manankalra , that is worse!

@ankit01ojha , no it is not following PEP8,

`{AnyOneOfRequirements(` is ok, but then all of the parts on subsequent lines would need to be aligned with the opening bracket, which is the parentheses (*round* bracket), not the brace `{`. 

I told you how to do it at https://github.com/coala/coala-bears/pull/2195#discussion_r179915624

Anyway, the issue has been re-assigned to someone else, as you are busy with GSoC at @openMF and we need this urgently.
